### PR TITLE
allow segment-specific `radius` and `inner_radius` for `pie`

### PR DESCRIPTION
**Pull request by ctarn**
This PR is based on #4027, and thus please don't merge it before #4027 is finished. [Update: Done]

I extend the `radius` and `inner_radius` of `pie` so that we now can easily plot the following figure:
```julia
import Makie, CairoMakie

fig = Makie.Figure()
ax = Makie.Axis(fig[1, 1]; autolimitaspect=1)

kw = (; offset_radius=0.4, strokecolor=:transparent, strokewidth=0)
Makie.pie!(ax, ones(7); radius=sqrt.(2:8) * 3, kw..., color=Makie.wong_colors(0.8)[1:7])

vs = [2, 3, 4, 5, 6, 7, 8]
vs_inner = [1, 1, 1, 1, 2, 2, 2]
rs = 8
rs_inner = sqrt.(vs_inner ./ vs) * rs

lp = Makie.LinePattern(; direction=Makie.Vec2f(1, -1), width=2, tilesize=(12, 12), linecolor=:darkgrey, background_color=:transparent)
# draw the inner pie twice since `color` can not be vector of `LinePattern` currently
Makie.pie!(ax, 20, 0, vs; radius=rs_inner, inner_radius=0, kw..., color=Makie.wong_colors(0.4)[eachindex(vs)])
Makie.pie!(ax, 20, 0, vs; radius=rs_inner, inner_radius=0, kw..., color=lp)
Makie.pie!(ax, 20, 0, vs; radius=rs, inner_radius=rs_inner, kw..., color=Makie.wong_colors(0.8)[eachindex(vs)])

display(fig)
```
pie-radius

### CODE REVIEW

**Comment by ctarn** [comment]
PR #4027 has been finished, and the PR is ready to be reviewed.
Is there anything I can do so you don't need to copy the PR before merging? @ffreyer Thank you!

**Comment by ctarn** [comment]
Yes! I have transfered the fork into my own account, and see the checkbox now. Hope it works now! Thank you!

File: ReferenceTests/src/tests/examples2d.jl [hunk 1]
```diff
@@ -373,6 +373,27 @@ end
...
+    rs = 8
+    rs_inner = sqrt.(vs_inner ./ vs) * rs
+
+    lp = Makie.LinePattern(; direction=Makie.Vec2f(1, -1), width=2, tilesize=(12, 12), linecolor=:darkgrey, background_color=:transparent)
+    # draw the inner pie twice since `color` can not be vector of `LinePattern` currently
+    pie!(ax, 20, 0, vs; radius=rs_inner, inner_radius=0, kw..., color=Makie.wong_colors(0.4)[eachindex(vs)])
+    pie!(ax, 20, 0, vs; radius=rs_inner, inner_radius=0, kw..., color=lp)
+    pie!(ax, 20, 0, vs; radius=rs, inner_radius=rs_inner, kw..., color=Makie.wong_colors(0.8)[eachindex(vs)])
+
+    display(fig)
```
Comment (ffreyer) [inline ReferenceTests/src/tests/examples2d.jl:394]: I think this is causing the test failures?
```suggestion
    fig
```

Comment (ctarn) [reply ReferenceTests/src/tests/examples2d.jl]: Thanks! fixed.

**Comment by ffreyer** [comment]
Something is wrong with docs here. 360m runtime...

**Comment by ctarn** [comment]
seems that similar issues appear in docs/src/reference/plots/pie.md :(
trying again

**Comment by ctarn** [comment]
The test has passed. Thanks!

**Pull request merged**